File: leetcode_70/solution.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    if (n <= 2)
        return n;
    var dp1 = 1, dp2 = 2;
    for (let i = 2; i < n; i++) {
        let dp = dp1 + dp2;
        dp1 = dp2;
        dp2 = dp;
    }
    return dp2;
};

File: leetcode_70/solution.go
package main

func climbStairs(n int) int {
	if n <= 2 {
		return n
	} else {
		dp1, dp2 := 1, 2
		for i := 2; i < n; i += 1 {
			dp1, dp2 = dp2, dp1 + dp2
		}
		return dp2
	}
}